src/estimation/src/GravityCompensationHelpers.cpp

```diff
@@ -0,0 +1,186 @@
+    {
+      m_model.computeFullTreeTraversal(*m_kinematicTraversals[floatingLinkIndex], floatingLinkIndex);
+    }
+    
+    // To initialize the kinmatic propagation, we should first convert the kinematics
+    // information from the frame in which they are specified to the main frame of the link
+    iDynTree::Transform link_H_frame = m_model.getFrameTransform(floatingFrame);
+    
+    iDynTree::Twist base_vel_frame, base_vel_link;
+    iDynTree::SpatialAcc base_acc_frame, base_acc_link;
+    iDynTree::ClassicalAcc base_classical_acc_link;
+    
+    iDynTree::Vector3 zero3;
+    zero3.zero();
+    
+    base_vel_frame.setLinearVec3(zero3);
+    base_vel_frame.setAngularVec3(zero3);
+    base_vel_link = link_H_frame*base_vel_frame;
+    
+    base_acc_frame.setLinearVec3(properClassicalLinearAcceleration);
+    base_acc_frame.setAngularVec3(zero3);
+    base_acc_link = link_H_frame*base_acc_frame;
+    
+    base_classical_acc_link.fromSpatial(base_acc_link, base_vel_link);
+    
+    // Propagate the kinematics information
+    bool ok = iDynTree::dynamicsEstimationForwardVelAccKinematics(m_model, *(m_kinematicTraversals[floatingLinkIndex]),
+                                                                  base_classical_acc_link.getLinearVec3(),
+                                                                  base_vel_link.getAngularVec3(),
+                                                                  base_classical_acc_link.getAngularVec3(),
+                                                                  jointPos, m_jointDofsZero, m_jointDofsZero,
+                                                                  m_linkVels, m_linkProperAccs);
+    // store joint positions
+    m_jointPos = jointPos;
+    
+    if (!ok)
+    {
+      return false;
+    }
+    else
+    {
+      m_isKinematicsUpdated = true;
+      return true;
+    }
+  }
+  
+  bool GravityCompensationHelper::getGravityCompensationTorques(JointDOFsDoubleArray& jointTrqs)
+  {
+    if (!m_isModelValid)
+    {
+      iDynTree::reportError("GravityCompensationHelper", "getGravityCompensationTorques", "Model not set");
+      return false;
+    }
+    
+    if (!m_isKinematicsUpdated)
+    {
+      iDynTree::reportError("GravityCompensationHelper", "getGravityCompensationTorques", "Kinematic information not set");
+      return false;
+    }
+    
+    // Compute joint torques
+    bool ok = iDynTree::RNEADynamicPhase(m_model, m_dynamicTraversal, m_jointPos, m_linkVels, m_linkProperAccs,
+                                         m_linkNetExternalWrenchesZero, m_linkIntWrenches, m_generalizedTorques);
+    if (!ok)
+    {
+      iDynTree::reportError("GravityCompensationHelper", "getGravityCompensationTorques", "Error in computing RNEADynamicPhase");
+      return false;
+    }
+    
+    // store computed torques
+    jointTrqs = m_generalizedTorques.jointTorques();
+    
+    return true;
+  }
+ 
+}
```

Comment:
Can you add a newline at the end of this file? Otherwise @francesco-romano will complain. :D